tools/cudaq-quake/cudaq-quake.cpp

```diff
@@ -305,7 +305,8 @@ int main(int argc, char **argv) {
   [[maybe_unused]] llvm::InitLLVM unused(argc, argv);
   cl::ParseCommandLineOptions(argc, argv, toolName);
   if (showVersion)
-    llvm::errs() << "nvq++ Version " << cudaq::getVersion() << '\n';
+    llvm::errs() << "nvq++ Version " << cudaq::getVersion() << " ("
+                 << cudaq::getFullRepositoryVersion() << ")" << '\n';
```

Comment:
What do you think about having this SHA information placed in the `nvq++` driver instead of in any of the executables?

Reply:
I don't have a strong opinion here :) Is there a reason for placing it in the `cudaq-quake` executable originally?

Reply:
> What do you think about having this SHA information placed in the `nvq++` driver instead of in any of the executables?

I suspect that wouldn't work with Python.

Reply:
Neither will placing it here, I think.

Reply:
Right, sorry. But it is also in the runtime, so the Python implementation will inherit it from there. So we could indeed pull it out of this location (and place it in `nvq++` driver script instead), but if we pull line 309 out, we should also pull line 308 (pre-existing) out, too.

As mentioned one of my other comments in this thread, this change doesn't cause any additional linking steps during regular developer development.

Reply:
I've revised this PR to have the version info in the `nvq++` driver script. Also, added a help string for `--version`, which we handle but didn't mention.

Reply:
I need to open an RFC on our option names in the driver. It's getting a bit murky. :)